reference-implementation/lib/readable-stream-abstract-ops.js

```diff
@@ -159,6 +159,42 @@ export function IsReadableStream(x) {
   return true;
 }
 
+export function PutBackIntoReadableStream(stream, chunk) {
+  if (stream._state === 'closed') {
+    throw new TypeError('stream is closed');
```

Comment:
Hmm, but that kind of sucks because what if you want to asynchronously make the decision to put it back or not?

Maybe we need `.read({ preventClose: true })`?? But then how do you close it later? `.cancel()` I guess??? Icky.